vm/devices/net/netvsp/src/lib.rs

```diff
@@ -813,11 +813,32 @@ impl PrimaryChannelState {
             .collect();
 
         let rss_state = rss_state
-            .map(|rss| {
+            .map(|mut rss| {
                 if rss.indirection_table.len() != indirection_table_size as usize {
-                    return Err(NetRestoreError::MismatchedIndirectionTableSize);
+                    // Dynamic reduction of indirection table can cause unexpected and hard to investigate issues
+                    // with performance and processor overloading. Nic should not reduce the indirection table during restore.
+                    // Logging a warning and continuing with the restore.
+                    tracing::warn!(
```

Comment:
I think we should still crash UH when the indirection table size is decreased as othewise continuing to run as if nothing has changed will make it difficult to troubleshoot performance issues